temporal-sdk/src/main/java/io/temporal/workflow/Workflow.java

```diff
@@ -615,6 +615,14 @@ public static WorkflowInfo getInfo() {
     return WorkflowInternal.getWorkflowInfo();
   }
 
+  public static <T> Object getMemo(String key, Class<T> valueClass) {
+    return getMemo(key, valueClass, valueClass.getGenericSuperclass());
+  }
+
+  public static <T> T getMemo(String key, Class<T> valueClass, Type valueType) {
```

Comment:
1. Rename valueType to genericType, it's confusing otherwise.
2. Please provide javadocs for both methods. It's also a good idea to link a doc to DataConverter#fromPayloads to explain why there are two parameters here. DataConverter#fromPayloads is well documented by itself.